lib/queue.js

```diff
@@ -1,27 +1,67 @@
+
+
 class Queue {
-    constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+    constructor(size = 25) {
+        this.size = size
+        this.store = new Array(size);
+        this.front = null;
+        this.rear = null;
     }
 
     enqueue(element) {
-        throw new Error("This method has not been implemented!");
+        if (this.front == 0 && (this.rear == this.store.size - 1)){
+            return false
+        } else if (this.front == null) {
+            // if the queue is empty
+            this.front = 0
+            this.rear = 0
+            this.store[this.rear] = element
+        } else {
+            // remainder gives us 0 when full
+            this.rear = (this.rear + 1) % this.size
+            this.store[this.rear] = element
+        }
+
     }
 
     dequeue() {
-        throw new Error("This method has not been implemented!");
+        if (this.front == null){
+            return false;
+        } else if (this.front == this.rear){
+            let element = this.store[this.front]
+            this.front = null
+            this.rear = null
+            return element 
+        } else {
+            let element = this.store[this.front]
+            this.store[this.front] = null
+            this.front = (this.front + 1 ) % this.size
+            return element
+        }
     }
 
-    front() {
-        throw new Error("This method has not been implemented!");
+    frontItem() {
+        console.log(this.front)
+        return this.store[this.front]
     }
 
-    size() {
-        throw new Error("This method has not been implemented!");
+    sizeOfQ() {
```

Comment:
👍 , although since this `size` method is a member of the `Queue` instance, the "Q" is kind of assumed.